components/security/authentication.rst

```diff
@@ -267,5 +267,55 @@ in) is correct, you can use::
         $user->getSalt()
     );
 
+Authentication Events
+---------------------
+
+The security component provides 4 related authentication events:
+
+===============================  ================================================  =========================================================================
+Name                             Event Constant                                    Argument Passed to the Listener
+===============================  ================================================  =========================================================================
+security.authentication.success  ``AuthenticationEvents::AUTHENTICATION_SUCCESS``  :class:`Symfony\Component\Security\Core\Event\AuthenticationEvent`
+security.authentication.failure  ``AuthenticationEvents::AUTHENTICATION_FAILURE``  :class:`Symfony\Component\Security\Core\Event\AuthenticationFailureEvent`
+security.interactive_login       ``SecurityEvents::INTERACTIVE_LOGIN``             :class:`Symfony\Component\Security\Http\Event\InteractiveLoginEvent`
+security.switch_user             ``SecurityEvents::SWITCH_USER``                   :class:`Symfony\Component\Security\Http\Event\SwitchUserEvent`
+===============================  ================================================  =========================================================================
+
+Authentication Success and Failure Events
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a provider authenticates the user, a ``security.authentication.success``
+event is dispatched.  Likewise, when no providers authenticate the user,
+a ``security.authentication.failure`` event is dispatched.  You
+could listen on the ``security.authentication.failure`` event, for example,
+in order to log failed login attempts.
+
+.. tip::
+
+    One of the authentication events is always triggered when a request points
+    to a secured area.
```

Comment:
As far as I can tell, I don't think this is true. I just tried with Symfony 3 and an authenticator implementing `SimplePreAuthenticatorInterface`. `security.authentication.success` is called only the first time you are authenticated. Thus, if you are already logged in, none of the authentication events will fire.